flytekit/deck/deck.py

```diff
@@ -0,0 +1,79 @@
+import os
+from typing import List, Union
+
+from jinja2 import Environment, FileSystemLoader
+
+from flytekit import ExecutionParameters, FlyteContext, FlyteContextManager
+from flytekit.core.type_engine import TypeEngine
+from flytekit.deck.renderer import Renderer
+
+DEFAULT_DECK_NAME = "default"
+
+
+class Deck:
+    # Deck enable users to get customizable and default visibility into their tasks.
+    #
+    # Deck contains a list of renderers (FrameRenderer, MarkdownRenderer) that can
+    # generate a html file. For example, FrameRenderer can render a DataFrame as an HTML table,
+    # MarkdownRenderer can convert Markdown string to HTML
+    #
+    # Flyte context saves a list of deck objects, and we use renderers in those decks to render
+    # the data and create an HTML file when those tasks are executed
+    #
+    # Each task has a least three decks (input, output, default). Input/output decks are
+    # used to render tasks' input/output data, and the default deck is used to render line plots,
+    # scatter plots or markdown text. In addition, users can create new decks to render
+    # their data with custom renderers.
+    def __init__(self, name: str, renderers: Union[List[Renderer], Renderer]):
+        self.name = name
+        self.renderers = renderers if isinstance(renderers, list) else [renderers]
+        FlyteContextManager.current_context().user_space_params.decks.append(self)
+
+    def append(self, r: Renderer) -> "Deck":
+        self.renderers.append(r)
+        return self
+
+
+def _output_deck(task_name: str, new_user_params: ExecutionParameters, task_input: dict, task_output: dict):
+    deck_map = {}
+    INPUT = "input"
+    OUTPUT = "output"
+    ctx = FlyteContext.current_context()
+    output_dir = ctx.file_access.get_random_local_directory()
+
+    deck_map[INPUT] = []
+    for k, v in task_input.items():
+        _deck_to_html_file(deck_map, INPUT, output_dir, k, TypeEngine.to_html(ctx, v, type(v)))
```

Comment:
So this also works for inputs, meaning if an input type has `to_html` implemented, it will be rendered in the deck, correct? We often have large specs passed into input fields, and rendering them in the task view would be a huge QoL improvement!